packages/common/nbstore/src/sync/awareness/index.ts

```diff
@@ -18,8 +18,11 @@ export class AwarenessSyncImpl implements AwarenessSync {
 
   async update(record: AwarenessRecord, origin?: string) {
     await Promise.all(
-      [this.storages.local, ...Object.values(this.storages.remotes)].map(peer =>
-        peer.update(record, origin)
+      [this.storages.local, ...Object.values(this.storages.remotes)].map(
+        peer =>
+          peer.connection.status === 'connected'
```

Comment:
connection status should be an internal state

Reply:
actually connection itself is internal

Reply:
But this often needs to know the status of the connection from outside